lib/Sema/CSSimplify.cpp

```diff
@@ -8097,6 +8097,26 @@ ConstraintSystem::matchTypes(Type type1, Type type2, ConstraintKind kind,
     return getTypeMatchSuccess();
   }
 
+  // If we're trying to match an 'Array' to a 'Slab', check to see if the
+  // locator is anchored to an array literal expression. If it is, we can
+  // interpret the array literal as a slab literal instead.
+  if (type1->isArray() && type2->isSlab()) {
+    // If the locator is pointing directly at an array literal, allow the
+    // conversion.
+    if (isExpr<ArrayExpr>(locator.getAnchor())) {
```

Comment:
And one more thing here - `locator.directlyAt<ArrayExpr>()` is what you want to use instead.